bentoml/yatai/python_api.py

```diff
@@ -149,3 +166,206 @@ def update_bento_upload_progress(
+
+
+# TODO update_deployment is not finished.  It will be working on along with cli command
+def update_deployment(deployment_name, namespace, yatai_service=None):
+    from bentoml.yatai import get_yatai_service
+
+    if yatai_service is None:
+        yatai_service = get_yatai_service()
+
+    # Make sure there is deployment with the same deployment name
+    get_deployment_pb = yatai_service.GetDeployment(
+        GetDeploymentRequest(deployment_name=deployment_name, namespace=namespace)
+    )
+    if get_deployment_pb.status.status_code == status_pb2.Status.NOT_FOUND:
+        raise BentoMLDeploymentException(
+            'Deployment {name} does not exist, please create deployment first'.format(
+                name=deployment_name
+            )
+        )
+
+
+def apply_deployment(deployment_info, yatai_service=None):
+    from bentoml.yatai import get_yatai_service
+
+    try:
+        if isinstance(deployment_info, dict):
+            deployment_pb = deployment_dict_to_pb(deployment_info)
+        elif isinstance(deployment_info, str):
+            deployment_pb = deployment_yaml_string_to_pb(deployment_info)
+        else:
+            raise BentoMLDeploymentException(
+                'Currently, we do not handle deployment info type {}'.format(
+                    str(type(deployment_info))
+                )
+            )
+
+        print('DDDD', deployment_pb)
+        validation_errors = validate_deployment_pb_schema(deployment_pb)
+        if validation_errors:
+            return ApplyDeploymentResponse(
+                status=Status.ABORTED(
+                    'Failed to validate deployment. {errors}'.format(
+                        errors=validation_errors
+                    )
+                )
+            )
+
+        if yatai_service is None:
+            yatai_service = get_yatai_service()
+        return yatai_service.ApplyDeployment(
+            ApplyDeploymentRequest(deployment=deployment_pb)
+        )
+    except BentoMLDeploymentException as error:
+        return ApplyDeploymentResponse(status=Status.ABORTED(str(error)))
+    except BentoMLException as error:
+        return ApplyDeploymentResponse(status=Status.INTERNAL(str(error)))
+
+
+def describe_deployment(namespace, name, yatai_service=None):
+    from bentoml.yatai import get_yatai_service
+
+    if yatai_service is None:
+        yatai_service = get_yatai_service()
+    return yatai_service.DescribeDeployment(
+        DescribeDeploymentRequest(deployment_name=name, namespace=namespace)
+    )
+
+
+def get_deployment(namespace, name, yatai_service=None):
+    from bentoml.yatai import get_yatai_service
+
+    if yatai_service is None:
+        yatai_service = get_yatai_service()
+    return yatai_service.GetDeployment(
+        GetDeploymentRequest(deployment_name=name, namespace=namespace)
+    )
+
+
+def delete_deployment(deployment_name, namespace, force_delete, yatai_service=None):
```

Comment:
make it default "force_delete=false"? which is consistant with yatai_service api's default behavior

Reply:
or default=None, works too